webhook_server_container/tests/test_github_api.py

```diff
@@ -157,36 +244,165 @@ def test_all_approvers_reviewers(process_github_webhook, approvers_and_reviewers
     assert all_reviewers == process_github_webhook.all_reviewers
 
 
-def test_check_if_pr_approved(process_github_webhook, approvers_and_reviewers, all_approvers_reviewers):
-    pr_approved_all_result = process_github_webhook._check_if_pr_approved(
+def test_check_pr_approved(process_github_webhook, approvers_and_reviewers, all_approvers_reviewers):
+    check_if_pr_approved = process_github_webhook._check_if_pr_approved(
         labels=[
-            f"{APPROVED_BY_LABEL_PREFIX}approver1",
-            f"{APPROVED_BY_LABEL_PREFIX}approver2",
-            f"{APPROVED_BY_LABEL_PREFIX}approver3",
-            f"{APPROVED_BY_LABEL_PREFIX}approver4",
+            f"{APPROVED_BY_LABEL_PREFIX}root_approver1",
+            f"{APPROVED_BY_LABEL_PREFIX}root_approver2",
+            f"{APPROVED_BY_LABEL_PREFIX}folder1_approver1",
+            f"{APPROVED_BY_LABEL_PREFIX}folder1_approver2",
+            f"{APPROVED_BY_LABEL_PREFIX}folder4_approver1",
+            f"{APPROVED_BY_LABEL_PREFIX}folder4_approver2",
+            f"{APPROVED_BY_LABEL_PREFIX}folder5_approver1",
+            f"{APPROVED_BY_LABEL_PREFIX}folder5_approver2",
         ]
     )
-    assert pr_approved_all_result == ""
+    assert check_if_pr_approved == ""
+
 
-    pr_approved_minimum_result = process_github_webhook._check_if_pr_approved(
+def test_check_pr_minimum_approved(process_github_webhook, approvers_and_reviewers, all_approvers_reviewers):
+    check_if_pr_approved = process_github_webhook._check_if_pr_approved(
         labels=[
-            f"{APPROVED_BY_LABEL_PREFIX}approver1",
-            f"{APPROVED_BY_LABEL_PREFIX}approver3",
+            f"{APPROVED_BY_LABEL_PREFIX}root_approver1",
+            f"{APPROVED_BY_LABEL_PREFIX}folder1_approver1",
+            f"{APPROVED_BY_LABEL_PREFIX}folder4_approver1",
+            f"{APPROVED_BY_LABEL_PREFIX}folder5_approver1",
         ]
     )
-    assert pr_approved_minimum_result == ""
+    assert check_if_pr_approved == ""
+
 
-    pr_not_approved_result = process_github_webhook._check_if_pr_approved(
+def test_check_pr_not_approved(process_github_webhook, approvers_and_reviewers, all_approvers_reviewers):
+    check_if_pr_approved = process_github_webhook._check_if_pr_approved(
         labels=[
-            f"{APPROVED_BY_LABEL_PREFIX}approver1",
+            f"{APPROVED_BY_LABEL_PREFIX}root_approver1",
         ]
     )
-    assert pr_not_approved_result == "Missing lgtm/approved from approvers: approver3, approver4\n"
+    assert not [
+        True
+        for appr in ["folder1_approver1", "folder1_approver2", "folder4_approver1", "folder4_approver2"]
+        if appr not in check_if_pr_approved
+    ]
```

Comment:
should not this include folder5_approver1 and folder5_approver2? ALL_CHANGED_FILES includes "folder5/file" and it has it's own owners file